extension.neon

```diff
@@ -189,6 +189,11 @@ services:
         tags:
             - phpstan.phpDoc.typeNodeResolverExtension
 
+    -
+        class: NunoMaduro\Larastan\Rules\NoUnnecessaryCollectionCallRule
```

Comment:
I definitely think this should be opt-in if it is going to be included. This rule is quite opinionated

Reply:
Here at Larastan everything is one big opt-in!

Reply:
This would be the first rule introduced to this codebase. Not everything needs to be done at the database layer, especially the likes of `firstWhere` if an index won't  be hit, unless I'm misunderstanding something.

Don't get me wrong, I'm sure there's lots of benefit and this is on the right track for sure. I do think there needs to be some customization or opt-in nature to it.

Reply:
I feel the same, could be too opinionated.

Opt-in? Opt-out? Enable by default, easy to disable?

Not too acquainted on the technical level with phpstan/larastan what the options are here (instead of simply not including and having to add it manually: which could hinder users of larastan even know about it)

Reply:
Usage of PHPStan goes basically like this.

1. you run phpstan
1. fix everything
1. Not possible to fix? Add it to `ignoreErrors:`

Reply:
Although we can add this particular rule on a higher Level

Reply:
That makes sense for actual issues, but this is quite opinionated and will likely have a lot of false positives depending on use cases

Reply:
@mr-feek Good point.

I added three new parameters to the config that allows you to customize the behavior of the rule:
```
noUnnecessaryCollectionCall: bool
noUnnecessaryCollectionCallOnly: string[]
noUnnecessaryCollectionCallExcept: string[]
```

This way a developer can easily enable or disable the rule. In addition, certain collection method calls can be ignored all together. E.g. if you only want to check `count` and `first`, you could set:
```
noUnnecessaryCollectionCallOnly: ['count', 'first']
```
in the configuration file. Similarly you can ignore specific method calls.

I'm not sure what sensible defaults would be though.

Enable the rule by default, but only check for `['count', 'first']` seems like it would already catch a fair amount of actual errors without being too opinionated. I'm also fine with disabling the rule altogether by default.

Reply:
awesome! I'd be totally onboard with enabling it by default.